spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/JdbcAggregateTemplateIntegrationTests.java

```diff
@@ -466,7 +461,38 @@ public void saveAndLoadLongChain() {
 		template.delete(chain4, Chain4.class);
 
 		String countSelect = "SELECT COUNT(*) FROM %s";
-		jdbcTemplate.queryForObject(String.format(countSelect, "CHAIN0"),emptyMap(), Long.class);
+		jdbcTemplate.queryForObject(String.format(countSelect, "CHAIN0"), emptyMap(), Long.class);
+	}
+
+	@Test // DATAJDBC-359
+	public void saveAndLoadLongChainWithoutIds() {
+
+		NoIdChain4 chain4 = new NoIdChain4();
```

Comment:
How about some more semantic naming (inner, middle, outer). Probably, also the numbering is confusing because I'd expect to start with zero with the outermost.

Reply:
The names are actually really useful when thinking about the generated SQL, since it is kind of recursive.

Maybe it makes it easier when one thinks as `Chain4` as shorthand for `Chain containing 4 elements`.
I added a comment to that extend.